packages/debounce/src/index.ts

```diff
@@ -13,20 +18,18 @@ import { onCleanup } from "solid-js";
  * fn('my-new-value');
  * ```
  */
-const createDebounce = <T extends (...args: unknown[]) => void, A = Parameters<T>>(
-  func: T,
+export function createDebounce<Args extends any[], F extends (...args: Args) => any>(
```

Comment:
I don't think the debounced function should allow for return type, we are not handling the returned type in any way, it will always return undefined.
https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXzDigxABEQAjHNMEARngB4AVeEADxNWAGd4AKAHTDYAcx4AueGgDWqHAHdUAbQC6ASngBeAHzwAbjizAANPACC2+AAVYUALYgSMHix07+AKHhIaU5ibe8ApQWBgA-FKoyPYUIDCe6lLKiKhSQiIw4lKWHFy80qhyiiqq8OEW8FEg+vGaugZGpgQQILDp9XqGxqqenqCQsAgo6Nh4BEQk5FQ0IABMTOZZfHkg3HxQqACeamYAYmycawUZgmKSFsud8JtbHkEjYFJ7gT4hYZHw0bHxic-wADJ4ABvFptGAdbRdJrwAC+fQG0DgvlGuHwhDaU0o1HQIAAzItlod8httmp7j5HulhGdljkrlDGsZXsFQhEojE4gkkgJTud6eJrt1gJogaDIODIQ1hXC+gB6OXwLA8HjIEBSAAOdkccBcBE28gw8Di8AwWw1IGAnjAeB4RuUoAodDMEtgZS0E0xZGxszo-H4aW+XOudDmePU1ttRrgPAYHsdftD4fliuVqvVNwgEEUfEQOBg8DgGGQMHwZotZgoyCNGAAFsMaGN8AosFnMyEtnwiyX8GhQIgsKhLZHUHb4I6Fh6McRvTNcXN-YHOXVGUmIzbR9GQDxJ+PKAu1yn4AArZBjmBYUS1o0SddRvcUAlTyaznG0PGLiRBlcNQ8b8-bk+D4fmuQA